### PR TITLE
PMM-4447 Update PMM server.

### PR DESCRIPTION
**Pull request by askomorokhov**
https://jira.percona.com/browse/PMM-4447

### CODE REVIEW

File: pmm-app/src/pmm-update-panel/module.ts [hunk 1]
```diff
@@ -197,30 +210,30 @@ export class PanelCtrl extends MetricsPanelCtrl {
...
-                $scope.isUpdateAvailable = true;
-                $scope.canBeReloaded = true;
-                $scope.errorMessage = PanelCtrl.ERRORS.UPDATE;
-            }
-        }).catch(() => {
-            this.reset($scope);
+             const data = response.data;
+             $scope.isUpdated = 'done' in data ? data.done : false;
+             $scope.updateLogOffset = 'log_offset' in data ? data.log_offset : 0;
+             $scope.output += '\n' + data.log_lines.join('\n');
```
Comment (AlekSi) [inline pmm-app/src/pmm-update-panel/module.ts]: It is better to add `\n` to the end.

File: pmm-app/src/pmm-update-panel/module.ts [hunk 2]
```diff
@@ -197,30 +210,30 @@ export class PanelCtrl extends MetricsPanelCtrl {
...
+             window.setTimeout(this.getLog.bind(this, $scope, $http), 500);
+        }).catch((err) => {
+             $scope.updateCntErrors++;
```
Comment (AlekSi) [inline pmm-app/src/pmm-update-panel/module.ts]: > Reset error counter when a non-error response is received.

File: pmm-app/src/pmm-update-panel/module.ts [hunk 3]
```diff
@@ -197,30 +210,30 @@ export class PanelCtrl extends MetricsPanelCtrl {
      * Send request for get info about update status
      */
     private getLog($scope, $http): void {
-        if (!$scope.logLocation.length) return;
-
+        if ($scope.updateCntErrors > 600) {
+            console.log('error: ', $scope.errorMessage, $scope.updateCntErrors);
+            $scope.updateFailed = true;
+            return;
```
Comment (AlekSi) [inline pmm-app/src/pmm-update-panel/module.ts]: Previous code also set canBeReloaded. I have no idea if that is needed :)

File: pmm-app/src/pmm-update-panel/module.ts (hunk 1 above)
Comment (askomorokhov) [reply pmm-app/src/pmm-update-panel/module.ts]: Fixed.

File: pmm-app/src/pmm-update-panel/module.ts (hunk 2 above)
Comment (askomorokhov) [reply pmm-app/src/pmm-update-panel/module.ts]: Good catch - removed it for debug and forgot to add back.

File: pmm-app/src/pmm-update-panel/module.ts (hunk 3 above)
Comment (askomorokhov) [reply pmm-app/src/pmm-update-panel/module.ts]: I have added it here https://github.com/percona/grafana-dashboards/blob/322928e21cfbdaad227b7286bfda6dd21220552b/pmm-app/src/pmm-update-panel/module.ts#L219

**Pull request merged**